machines/arch_filelist.mk

```diff
@@ -0,0 +1,118 @@
+# This file contains a list of the *core* manycore architecture files that are
+# used in simulation. This has NOT been used in tapeout or any tapeout related
+# activities beyond simulation
+
+ARCH_VINCDIRS =$(BASEJUMP_STL_DIR)/bsg_misc
+ARCH_VINCDIRS+=$(BASEJUMP_STL_DIR)/bsg_cache
+ARCH_VINCDIRS+=$(BASEJUMP_STL_DIR)/bsg_noc
+ARCH_VINCDIRS+=$(BSG_MANYCORE_DIR)/v
+ARCH_VINCDIRS+=$(BSG_MANYCORE_DIR)/v/vanilla_bean
+
+ARCH_VDEFINES =$(BASEJUMP_STL_DIR)/bsg_misc/bsg_defines.v
```

Comment:
What do you mean by ARCH_VDEFINES? Do you mean VSOURCES instead?

Reply:
bsg_defines.v has to be at the top of the list of files to be compiled.

Reply:
> We have two types of `.v` files in basejump_stl. `bsg_defines.v` is one kind with defines and `bsg_dff_reset.v` is the other kind with actual rtl. Hence, I used `VDEFINES` for `.v` with defines and `MACRO_DEFS` for cmdline macro defs.
@drichmond 

@tommydcjung! This is one of the reasons I used make variables instead `-f` option of vcs. We can group `bsg_defines.v` & co in `ARCH_VDEFINES` make variable. Files in `ARCH_VDEFINES` are compiled first:
https://github.com/bespoke-silicon-group/bsg_manycore/blob/1f653f7532fbdd19e054c0fc96c942dc97f2d7ac/machines/Makefile#L32